azurerm/internal/services/netapp/netapp_volume_data_source.go

```diff
@@ -75,6 +75,32 @@ func dataSourceNetAppVolume() *schema.Resource {
 				Computed: true,
 				Elem:     &schema.Schema{Type: schema.TypeString},
 			},
+
+			"data_protection_replication": {
+				Type:     schema.TypeList,
+				Computed: true,
+				MaxItems: 1,
```

Comment:
```suggestion
```

Reply:
If I remove MaxItems: 1, will this enforce the same rule of not allowing more than one "data_protection_replication" ?

Reply:
I looked at options like AtLeastOneOf and ExactOneOf by they will make this attribute required and we can't have it, maybe I'm thinking if we can extend the class that implements these and include a third option, e.g. WhenInUseExactOneOf. Thoughts on that?

Reply:
If that is ok, I forked the Terraform SDK and committed what I think it is the implementation for that, let me know if this is a good idea and I can move on with adding the testcases and place a PR on the SDK. Is there a way to test this on my provider repo? Wanted to run some tests before the PR on SDK. I changed my local  vendor\github.com\hashicorp\terraform-plugin-sdk\helper\schema\schema.go but this attribute still cannot be recognized.

![image](https://user-images.githubusercontent.com/17993275/108435081-fc67dc00-7205-11eb-9d1b-2ed3c1e6ce56.png)

Reply:
Just realized I didn't paste the sdk commit for you to check: https://github.com/paulomarquesc/terraform-plugin-sdk/commit/73b5f2b1a37d2e49f23c3738ea01ddc16c12bf4a